tdrs-backend/tdpservice/email/email.py

```diff
@@ -12,27 +14,79 @@
+):
+    """Send an email to a user when their account approval status is updated."""
+    from tdpservice.users.models import AccountApprovalStatusChoices
+
+    template_path = None
+    subject = None
+    text_message = None
+    logger.info(f"Preparing email to {recipient_email} with status {new_approval_status}")
+    match new_approval_status:
+        case AccountApprovalStatusChoices.INITIAL:
+            # Stubbed for future use
+            return
+
+        case AccountApprovalStatusChoices.ACCESS_REQUEST:
+            template_path = EmailType.ACCESS_REQUEST_SUBMITTED.value
+            subject = 'Access Request Submitted'
+            text_message = 'Your account has been requested.'
+
+        case AccountApprovalStatusChoices.PENDING:
+            # Stubbed for future use
+            return
+
+        case AccountApprovalStatusChoices.APPROVED:
+            template_path = EmailType.REQUEST_APPROVED.value
+            subject = 'Access Request Approved'
+            text_message = 'Your account request has been approved.'
+
+        case AccountApprovalStatusChoices.DENIED:
+            template_path = EmailType.REQUEST_DENIED.value
+            subject = 'Access Request Denied'
+            text_message = 'Your account request has been denied.'
+
+        case AccountApprovalStatusChoices.DEACTIVATED:
+            template_path = EmailType.ACCOUNT_DEACTIVATED.value
+            subject = 'Account is Deactivated'
+            text_message = 'Your account has been deactivated.'
+
+    context.update({'subject': subject})
+
+    automated_email.delay(
+        email_path=template_path,
+        recipient_email=recipient_email,
+        subject=subject,
+        email_context=context,
+        text_message=text_message
+    )
+
+
 @shared_task
-def mail(email_path, recipient_email, email_context):
-    """Send an automated email to a user. Use mail.delay() to send asynchronously."""
-    subject = email_context["subject"]
+def automated_email(email_path, recipient_email, subject, email_context, text_message):
+    """Send email to user."""
+    logger.info(f"Starting celery task to send email to {recipient_email}")
     html_message = construct_email(email_path, email_context)
 
-    if email_context["text_message"]:
-        text_message = email_context["text_message"]
-    else:
-        text_message = "An email was sent with HTML content. Please view in an HTML capable email client."
     send_email(subject, text_message, html_message, [recipient_email])
 
 
 def construct_email(email_path, context):
     """Get email template."""
+    logger.info(f"Constructing email from template {email_path}")
     template = get_template(email_path)
+    logger.info(f"Email template rendered from the path {email_path}")
+
     return template.render(context)
 
 
 def send_email(subject, message, html_message, recipient_list):
     """Send an email to a list of recipients."""
+    logger.info(f"Envoked send_email with subject {subject}")
```

Comment:
some combination of line 89 and 105 would be useful as a logentry [ref](https://github.com/raft-tech/TANF-app/blob/0707d063a8d943fea682ba6d54d870699571fd64/tdrs-backend/tdpservice/security/models.py#L57)

Reply:
my solution might have been overkill but hopefully it's useful for some of the other email tickets as well. there's a new `log` function that logs to the system output and optionally creates a `LogEntry`. Emails aren't associated with a model, so i just used the user the email was intended for.